java/source/cdata.rst

```diff
@@ -0,0 +1,507 @@
+    print(type(py_array))
+    print(py_array)
+
+.. code-block:: shell
+
+    <class 'pyarrow.lib.Int32Array'>
+    [
+      1,
+      7,
+      93
+    ]
+
+Sharing VectorSchemaRoot
+************************
+
+Java Side:
+
+.. testcode::
+
+    import org.apache.arrow.c.ArrowArray;
+    import org.apache.arrow.c.ArrowSchema;
+    import org.apache.arrow.c.Data;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.util.AutoCloseables;
+    import org.apache.arrow.vector.IntVector;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+
+    import static java.util.Arrays.asList;
+
+    public class ShareVectorSchemaRootAPI {
+      final static BufferAllocator allocator = new RootAllocator();
+      final static Field column_one = new Field("column-one", FieldType.nullable(new ArrowType.Int(32, true)), null);
+      final static Schema schema = new Schema(asList(column_one));
+      final static VectorSchemaRoot root = VectorSchemaRoot.create(schema, allocator);
+
+      public static BufferAllocator getAllocatorForJavaConsumers() {
+        return allocator;
+      }
+
+      public static VectorSchemaRoot getVectorSchemaRootForJavaConsumers() {
+        IntVector intVector = (IntVector) root.getVector(0);
+        root.allocateNew();
+        intVector.set(0, 100);
+        intVector.set(1, 20);
+        root.setRowCount(2);
+        return root;
+      }
+
+      public static void closeAllocatorForJavaConsumers() {
+        try {
+          AutoCloseables.close(AutoCloseables.iter(root));
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      public static void simulateAsAJavaConsumers() {
+        try (ArrowArray arrowArray = ArrowArray.allocateNew(allocator);
+             ArrowSchema arrowSchema = ArrowSchema.allocateNew(allocator)
+        ) {
+          Data.exportVectorSchemaRoot(allocator, getVectorSchemaRootForJavaConsumers(), null, arrowArray, arrowSchema);
+          try (VectorSchemaRoot root = Data.importVectorSchemaRoot(allocator, arrowArray, arrowSchema, null);) {
+            System.out.print(root.contentToTSVString());
+          }
+        }
+        closeAllocatorForJavaConsumers();
+        allocator.close();
+      }
+    }
+
+    ShareVectorSchemaRootAPI.simulateAsAJavaConsumers();
+
+.. testoutput::
+
+    column-one
+    100
+    20
+
+Python Side:
+
+.. code-block:: python
+
+    import jpype
+    import pyarrow as pa
+    from pyarrow.cffi import ffi
+
+    jvmargs=["-Darrow.memory.debug.allocator=true"]
+    jpype.startJVM(*jvmargs, jvmpath=jpype.getDefaultJVMPath(), classpath=[
+        "./target/java-python-by-cdata-1.0-SNAPSHOT-jar-with-dependencies.jar"])
```

Comment:
It would be nice to leave a comment regarding the jar we have mentioned here.

Reply:
Added